runtime/description.js

```diff
@@ -177,7 +177,7 @@ export class DescriptionFormatter {
     // "A and b."
     // "A, b, ..., and z." (Oxford comma ftw)
     let delim = ['', '', ' and ', ', and '][Math.min(3, count)];
-    return nonEmptyStrings.slice(0, -1).join(', ') + delim + strings.pop();
+    return nonEmptyStrings.slice(0, -1).join(', ') + delim + nonEmptyStrings[count - 1];
```

Comment:
Consider
```
const lastString = nonEmptyStrings.pop();
return `${nonEmptyStrings.join(', ')}${delim}${lastString}`;
```

Also, I'm still not a fan of `!!`. Your filter above is semantically equivalent to `(str => str)`, but if you want to cast the return value (for reader clarity?) I recommend `Boolean(str)`.

Reply:
done